econsa/tests/test_correlation.py

```diff
@@ -0,0 +1,92 @@
+"""Test for correlation transformation."""
+import chaospy as cp
+import numpy as np
+import pytest
+from scipy.stats import norm
+
+from econsa.correlation import gc_correlation
+
+
+def get_strategies(name):
+    dim = np.random.randint(2, 5)
+    means = np.random.uniform(-100, 100, dim)
+
+    if name == "test_gc_correlation":
+        # list of distributions to draw from
+        # repeated distributions are for higher drawn frequency, not typo
+        distributions = [
+            cp.Normal,
+            cp.Uniform,
+            cp.Uniform,
+            cp.LogNormal,
+            cp.LogNormal,
+            cp.Exponential,
+            cp.Rayleigh,
+            cp.LogWeibull,
+        ]
+
+        marginals = list()
+        for mean in means:
+            dist_i = np.random.choice(len(distributions))
+            dist = distributions[dist_i]
+            marginals.append(dist(mean))
+
+        # redraw corr until is positive definite
+        while True:
+            corr = np.random.uniform(-1, 1, size=(dim, dim))
```

Comment:
We can write this more concise. Just sample a covariance matrix and then use our _cov2corr function.